Alpaca.Markets/PolygonSockClient.cs

```diff
@@ -0,0 +1,274 @@
+            String symbol)
+        {
+            subscribe($"T.{symbol}");
+        }
+
+        /// <summary>
+        /// Subscribes for the quote updates via <see cref="QuoteReceived"/>
+        /// event for specific asset from Polygon streaming API.
+        /// </summary>
+        /// <param name="symbol">Asset name for subscription change.</param>
+        public void SubscribeQuote(
+            String symbol)
+        {
+            subscribe($"Q.{symbol}");
+        }
+
+        /// <summary>
+        /// Subscribes for the second bar updates via <see cref="SecondAggReceived"/>
+        /// event for specific asset from Polygon streaming API.
+        /// </summary>
+        /// <param name="symbol">Asset name for subscription change.</param>
+        public void SubscribeSecondAgg(
+            String symbol)
+        {
+            subscribe($"A.{symbol}");
+        }
+
+        /// <summary>
+        /// Subscribes for the minute bar updates via <see cref="MinuteAggReceived"/>
+        /// event for specific asset from Polygon streaming API.
+        /// </summary>
+        /// <param name="symbol">Asset name for subscription change.</param>
+        public void SubscribeMinuteAgg(
+            String symbol)
+        {
+            subscribe($"AM.{symbol}");
+        }
+
+        /// <summary>
+        /// Unsubscribes from the trade updates via <see cref="TradeReceived"/>
+        /// event for specific asset from Polygon streaming API.
+        /// </summary>
+        /// <param name="symbol">Asset name for subscription change.</param>
+        public void UnsubscribeTrade(
+            String symbol)
+        {
+            unsubscribe($"T.{symbol}");
+        }
+
+        /// <summary>
+        /// Unsubscribes from the quote updates via <see cref="QuoteReceived"/>
+        /// event for specific asset from Polygon streaming API.
+        /// </summary>
+        /// <param name="symbol">Asset name for subscription change.</param>
+        public void UnsubscribeQuote(
+            String symbol)
+        {
+            unsubscribe($"Q.{symbol}");
+        }
+
+        /// <summary>
+        /// Unsubscribes from the second bar updates via <see cref="SecondAggReceived"/>
+        /// event for specific asset from Polygon streaming API.
+        /// </summary>
+        /// <param name="symbol">Asset name for subscription change.</param>
+        public void UnsubscribeSecondAgg(
+            String symbol)
+        {
+            unsubscribe($"A.{symbol}");
+        }
+
+        /// <summary>
+        /// Unsubscribes from the minute bar updates via <see cref="MinuteAggReceived"/>
+        /// event for specific asset from Polygon streaming API.
+        /// </summary>
+        /// <param name="symbol">Asset name for subscription change.</param>
+        public void UnsubscribeMinuteAgg(
+            String symbol)
+        {
+            unsubscribe($"AM.{symbol}");
+        }
+
+        private void subscribe(
```

Comment:
I have an idea! What if we will pass 2 strings into this method - `messageType` and symbol and will format topic inside this method. There are some reason for this change:

- we reduce number of places where we format topic string
- we can move all `messageType` parameter values into constants and re-use them in `handleMessageReceived` method

Reply:
That'd be fine, but how do you want to handle the enum? Since it's not going to be converted to a string by aJSON serializer directly, the enum values would basically just need to be `T, Q, AM, A` anyhow, so I don't really see a benefit. (Unless you want to do all the stuff described in the answers [here](https://stackoverflow.com/questions/27372816/how-to-read-the-value-for-an-enummember-attribute) to give them more descriptive names and get the attribute value out - I'm not a fan, though. )

Reply:
I guess you said use a constant rather than an enum - I suppose that's doable.